create_commits.py

```diff
@@ -0,0 +1,525 @@
+                "script. This is either because something went wrong during commit creation, or "
+                "because someone else is pushing to the remote branch as well. Aborting."
+                f"We pushed {len(remote_commit_hashes_created)} commits to the remote branch, "
+                f"with hashes, {remote_commit_hashes_created}. The local commits that we didn't "
+                f"push were {new_commit_local_hashes[len(remote_commit_hashes_created):]}."
+            )
+
+        # Create a commit on the remote branch, and store the OID of the created commit in
+        # last_commit_pushed
+        last_commit_pushed = create_commit_on_remote_branch(
+            github_token=github_token,
+            repository_name_with_owner=repository_name_with_owner,
+            remote_branch_name=remote_branch_name,
+            expected_head_oid=last_commit_pushed or merge_base_commit_oid,
+            file_changes=file_changes,
+            message=commit_message,
+        )
+        remote_commit_hashes_created.append(last_commit_pushed)
+        logging.info(
+            "Created commit %s from commit sha %s on branch %s/%s with message: %s",
+            last_commit_pushed,
+            local_commit_hash,
+            remote_name,
+            remote_branch_name,
+            commit_message,
+        )
+
+    logging.info(
+        "Finished creating %s commits on the remote branch %s/%s from the local branch %s.",
+        len(new_commit_local_hashes),
+        remote_name,
+        remote_branch_name,
+        local_branch_name,
+    )
+
+
+def validate_branch_name(
+    branch_name: str,
+    branch_type: Literal["remote", "local"],
+    remote_name: str = "origin",
+) -> None:
+    """
+    Validate that the branch name is provided without refs/heads/, and without the remote name. If
+    the branch name is invalid, raise a ValueError.
+    """
+
+    if branch_name.startswith("origin/"):
+        raise ValueError(f"Do not include 'origin/' in the {branch_type} branch name.")
+    if branch_name.startswith("refs/heads/"):
+        raise ValueError(
+            f"Do not include 'refs/heads/' in the {branch_type} branch name."
+        )
+    if branch_name.startswith(remote_name + "/"):
+        raise ValueError(
+            f"Do not include the remote name in the {branch_type} branch name."
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("github_token", help="GitHub personal access token", type=str)
+    parser.add_argument(
+        "repository_name_with_owner", help="Repository name with owner", type=str
+    )
+    parser.add_argument("local_branch_name", help="Local branch name", type=str)
+    parser.add_argument("remote_name", help="Remote name")
+    parser.add_argument("remote_branch_name", help="Remote branch name")
+    parser.add_argument("log_level", default="WARN", help="Log level", type=str)
```

Comment:
CLI convention is broadly speaking kebab-style (and we're weird at asana by deviating from it); this will still result in values like `args.github_token`, because `argparse` recognizes this
```suggestion
    parser.add_argument("github-token", help="GitHub personal access token", type=str)
    parser.add_argument(
        "repository-name-with-owner", help="Repository name with owner", type=str
    )
    parser.add_argument("local-branch-name", help="Local branch name", type=str)
    parser.add_argument("remote-name", help="Remote name")
    parser.add_argument("remote-branch-name", help="Remote branch name")
    parser.add_argument("log-level", default="WARN", help="Log level", type=str)
```